src/blueprints/tv2/factories/tv2-audio-action-factory.ts

```diff
@@ -6,30 +6,84 @@ import { PieceLifespan } from '../../../model/enums/piece-lifespan'
 import { TransitionType } from '../../../model/enums/transition-type'
 import { DeviceType } from '../../../model/enums/device-type'
 import { PieceActionType } from '../../../model/enums/action-type'
+import { Tv2AudioTimelineObjectFactory } from '../value-objects/factories/tv2-audio-timeline-object-factory'
+import { Tv2BlueprintConfiguration } from '../value-objects/tv2-blueprint-configuration'
 
 export class Tv2AudioActionFactory {
+  constructor(private readonly audioTimelineObjectFactory: Tv2AudioTimelineObjectFactory) { }
 
-  public createAudioActions(): Action[] {
+  public createAudioActions(blueprintConfiguration: Tv2BlueprintConfiguration): Action[] {
     return [
-      this.createStopAudioBedAction()
+      this.createStopAudioBedAction(),
+      this.createMicrophoneUpAction(blueprintConfiguration),
+      this.createMicrophoneDownAction(blueprintConfiguration)
     ]
   }
 
-  private createStopAudioBedAction(): PieceAction {
-    const duration: number = 1000
+  private createMicrophoneUpAction(blueprintConfiguration: Tv2BlueprintConfiguration): PieceAction {
+    const duration: number = 0
     const pieceInterface: PieceInterface = {
-      id: 'stopAudioBedPiece',
-      name: 'Stop audio bed',
+      ...this.createDefaultAudioPieceInterface(),
+      id: 'microphoneUpPiece',
+      name: 'Microphone Up',
+      layer: Tv2SourceLayer.AUDIO_ACTION_COMMAND,
+      duration,
+      tags: [],
+      timelineObjects: [
+        this.audioTimelineObjectFactory.createMicrophoneUpTimelineObject(blueprintConfiguration)
+      ]
+    }
+    return {
+      id: 'microphoneUpAction',
+      name: 'Microphone Up',
+      type: PieceActionType.INSERT_PIECE_AS_ON_AIR,
+      data: pieceInterface
+    }
+  }
+
+  private createDefaultAudioPieceInterface(): PieceInterface {
+    return {
       partId: '',
       type: PieceType.AUDIO,
-      layer: Tv2SourceLayer.AUDIO_BED,
       pieceLifespan: PieceLifespan.WITHIN_PART,
       transitionType: TransitionType.NO_TRANSITION,
       isPlanned: false,
       start: 0,
-      duration,
       preRollDuration: 0,
       postRollDuration: 0,
+    } as PieceInterface
```

Comment:
You should `as` when creating/instantiating interfaces/entities as you run the risk of violating the internal state, but simply not given the needed data.

Instead of using `as` here, just write the object out with "default" values. They are going to be overridden anyway when you do:
```
const pieceInterface: PieceInterface = { 
  ...this.createDefaultAudioPieceInterface,
  id: 'idOfAction'
  ...
}
```

Reply:
Has been updated for the commented on file, and the graphic file too.